### ISSUE TITLE
Pause video playback upon external events. [UX] [Improvement]

### ISSUE THREAD

**Issue by un-logic** [mu-fazil-vk/FluxTube#37]
This is a request for improving user experience of video playback.
**What**
As of version 0.7.0 whenever you receive a phone call or if your Bluetooth headset gets disconnected video playback will continue.
**Why**
With receiving incoming call, you get a problem of needing to rewind the content after you have ended the call or to pause playback manually during the call. This may happen if your caller application has function of a popup window instead of grabbing the focus entirely.
In case of disconnected Bluetooth device result is audio being rerouted to play through internal speakers. Which is undesirable, if you simply got out of range while listening or Bluetooth device's battery has died suddenly.
**How**
App should monitor events described previously, and may have option implemented to resume automatically upon phone call end or audio device reconnection.

**Comment by hafijulali** [comment]
Implementing this feature would require Phone permissions on Android. Which is one extra permission. Rather we may pause the video manually if phone starts ringing. 

We'll wait for sometime to see if other users want this feature to be implemented or not.

**Comment by un-logic** [comment]
1. You may not be required for Phone permission to be granted to implement automatic pause, others did that without. See screenshot of
![NewPipe](https://github.com/user-attachments/assets/63b9c2af-e482-4339-b052-4b2fb2f8d20f)
2. Headset disconnect should not require this permition to implement autopause either.

**Comment by hafijulali** [comment]
Yeah you're right. Cool, let me work on this hopefully we should see this in next update.

**Comment by un-logic** [comment]
For your information, iFrame player, introduced in v. 0.7.1, does stop upon any interruption described above.
